File: common_letters.R
# Function to get common characters
get_common_characters <- function(words) {
  # Initialize the list with characters from the first word
  common_chars <- strsplit(words[[1]], "")[[1]]
  
  # Find the intersection of all lists
  for (i in 2:length(words)) {
    word_chars <- strsplit(words[[i]], "")[[1]]
    common_chars <- intersect(common_chars, word_chars)
  }
  
  common_chars
}

# Main code
words <- readLines(con = "stdin")
common_chars <- get_common_characters(words)

# Sort and print the characters in alphabetical order without spaces
cat(sort(common_chars, decreasing = FALSE), sep = "")

File: common_letters.lua
local function updateLetterCount(word, letterCount)
    for i = 1, #word do
        local ch = word:sub(i, i)
        if ch:match("%a") then
            letterCount[ch] = (letterCount[ch] or 0) + 1
        end
    end
end

local letterCount = {}
local numWords = 0

-- Read words from input
while true do
    local word = io.read()
    if not word then break end
    updateLetterCount(word, letterCount)
    numWords = numWords + 1
end

-- Print common letters in alphabetical order
for code = string.byte('a'), string.byte('z') do
    local ch = string.char(code)
    if letterCount[ch] == numWords then
        io.write(ch)
    end
end
io.write("\n")
